### ISSUE TITLE
[$250] [HOLD for payment 2024-06-20] [HOLD for payment 2024-06-18] [Track Tax] Support tax tracking for split requests with workspace

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#39690]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.60-8
**Reproducible in staging?:** Yes
**Reproducible in production?:** No, new feature
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
Precondition:
- User is an employee of Collect workspace.
- The Collect workspace has several tax rates.

1. Go to staging.new.expensify.com
2. Go to workspace chat.
3. Create a split scan request with an invalid receipt and include a tax rate.
4. When the scanning fails, go to split preview.
5. Manually enter amount and merchant and save it with a tax rate.
6. Go to transaction thread of the split request.

## Expected Result:
The selected tax rate will show up under Tax rate row in transaction thread.

## Actual Result:
Tax rate row is blank in transaction thread, unless tax amount is edited.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/d80f0e0e-70e5-4739-8663-110f7d1b153f

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0126969ac729dd1083</li>
        <li>Upwork Job ID: 1776200720994766848</li>
        <li>Last Price Increase: 2024-06-20</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @dukenv0307</details>

**Comment by izarutskaya** [comment]
@garrettmknight I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by izarutskaya** [comment]
We think this issue might be related to the #collect project.

**Comment by MonilBhavsar** [comment]
Taking a look...

**Comment by MonilBhavsar** [comment]
Demoting from deploy blocker, because it's not an app blocker. This bug occurs because server is sending `taxCode` as null initially. We can hide the field at the frontend, but that won't be real fix for the issue. Since this is behind policyExpensechat beta, I think it's fine.
We're rolling out changes incrementally and tax support for split bill case is yet to be released. This issue will be fixed when `SplitBill` command accepts taxCode param and sends it back

**Comment by MonilBhavsar** [comment]
I'll start working soon...

**Comment by MonilBhavsar** [comment]
Finishing support for distance requests and then moving here

**Comment by MonilBhavsar** [comment]
Working on PR's now

**Comment by MonilBhavsar** [comment]
Auth and Web-E PR's are in review
Eto Olei shall be working on the frontend implementation

**Comment by MonilBhavsar** [comment]
Internal PR's are merged and deployed.
We now need to finish the App PR https://github.com/Expensify/App/pull/40240 to finally fix this issue.
The App PR is on HOLD for optimistic tax updates issue/PR https://github.com/Expensify/App/issues/39616

**Comment by MonilBhavsar** [comment]
The PR this issue depends is in review https://github.com/Expensify/App/issues/39616
Me or Eto Olei should get this next week.

**Comment by MonilBhavsar** [comment]
Same update as Friday^

**Comment by MonilBhavsar** [comment]
Working on the PR in another issue https://github.com/Expensify/App/issues/39616

**Comment by MonilBhavsar** [comment]
> Working on the PR in another issue https://github.com/Expensify/App/issues/39616

Same^
We'll circle back once that PR is merged

**Comment by MonilBhavsar** [comment]
Same^. Very close to merge the PR

**Comment by MonilBhavsar** [comment]
PR is ready for review

**Comment by garrettmknight** [comment]
Not overdue - still working through the PR.

**Comment by MonilBhavsar** [comment]
PR was deployed and reverted because of few blockers. I need to raise the PR again while fixing those blockers

**Comment by MonilBhavsar** [comment]
PR is in review again!

**Comment by garrettmknight** [comment]
@dukenv0307 offer is out to you. 50% x $250 for a regression = $125
[Upwork offer](https://www.upwork.com/nx/wm/offer/102813650)

**Comment by garrettmknight** [comment]
All set with payment - @dukenv0307 any regression test necessary?

**Comment by garrettmknight** [comment]
Ah, missed the double post. Closing!
